Adapt existing settings to the new security model

jpetrell opened the pull request:
Don't merge, requires custom image as Venus side changes have not been integrated yet.

![image](https://github.com/victronenergy/gui-v2/assets/2203667/0a86a238-4808-473d-a298-49925ce95323)

Changes made to gui-v1:

General
- New Security Profile setting: Secured, Weak, Unsecured

VRM
- "Logging enabled" on/off switch dropped
- New "VRM portal" setting: Off, Read-only, Full
- "VRM two-way communication" switch dropped

Remote Console
- "Manually reboot the GX device after changing these settings.\n\nFirst time use? Make sure to either set or disable the password check." warning dropped
- "Disable password check" and "Enable password check" buttons dropped
- "Enable on VRM" switch dropped
- "Enable on LAN" switch renamed to "Enable Remote Console"
- "Remote Console on VRM - status" moved at the end of the list

Services:
- "MQTT on LAN (SSL)" switch renamed to "MQTT Access"
- "MQTT on LAN (Plain-text)" switch dropped

Contributes to #811.

chriadam commented on pages/settings/PageSettingsLogger.qml:
```diff
@@ -34,13 +34,16 @@ Page {
-					{ display: CommonWords.enabled,	value: 1 },
+					{ display: CommonWords.off, value: VenusOS.Vrm_PortalMode_Off },
+					//% "Read-only"
+					{ display: qsTrId("settings_vrm_portal_readonly"), value: VenusOS.Vrm_PortalMode_ReadOnly },
```
We don't currently support this, see #560 but I guess since on VRM the button won't show up to launch Gui-v2 unless the user has full access, it's not a problem.

chriadam commented on pages/settings/PageSettingsLogger.qml:
```diff
@@ -150,12 +159,6 @@ Page {
 			}
 
-			ListSwitch {
-				//% "VRM two-way communication"
```
ooi what replaces this?  if this setting is disabled, a user can connect to VRM via their browser, but won't see any data (since the device isn't in contact with VRM to provide it).  I hope that we don't show the "launch gui-v2" button in VRM if this setting (or its new replacement) is false?

jpetrell replied:
VRM Portal mode setting on the same page, which need to be set to "Full".

jpetrell commented:
Adapted to Serj's input:
- In the list with radio buttons, it is good to show input field before interaction happens
- Confirmation dialog could be adjusted a bit, the bullet list to center align
- Moved the confirmation step to the end

Also, dropped "existing connections will be disconnected" warning. Jeroen commented that in most cases gui-v2 should automatically reconnect after the security profile change. The original warning was mainly added to gui-v1, because VNC disconnects completely. Though might be good idea to give warning when user switches from unsecured http to secured, since then the user needs to switch to https to reconnect.

Still needs more testing, Jeroen will build test image for people to comment on.

chriadam commented on pages/settings/PageSettingsLogger.qml:
```diff
@@ -76,20 +79,26 @@ Page {
 				]
 				dataItem.uid: Global.systemSettings.serviceUid + "/Settings/Vrmlogger/LogInterval"
-				allowed: !!loggerMode.dataItem.value && loggerMode.dataItem.value > 0
+				allowed: vrmPortalMode.dataItem.isValid && vrmPortalMode.dataItem.value > 0
```
maybe add a comment `// i.e. not Vrm_PortalMode_Off`

chriadam reviewed (approved):
Code LGTM, haven't tested on device as I believe it's still waiting for something in venus?

jpetrell commented:
> Code LGTM, haven't tested on device as I believe it's still waiting for something in venus?

You can install the custom image Jeroen provided using a USB stick.

chriadam commented:
Obsoleted by https://github.com/victronenergy/gui-v2/pull/1370, closing.
